app/serializers/post_serializer.rb

```diff
@@ -180,6 +181,10 @@ def can_recover
     scope.can_recover_post?(object)
   end
 
+  def can_see_hidden_post
+    scope.can_see_hidden_post?(object)
+  end
+
```

Comment:
Can you help add a test within this context, at the end: https://github.com/discourse/discourse/blob/5f1e1829563c6083426199e475535b78a3f581bb/spec/serializers/post_serializer_spec.rb#L133

I've written it here already, so just add it in another commit.

```
      it "includes if the user can see it" do
        expect(serialized_post_for_user(Fabricate(:moderator))[:can_see_hidden_post]).to eq(true)
        expect(serialized_post_for_user(Fabricate(:admin))[:can_see_hidden_post]).to eq(true)
        expect(serialized_post_for_user(user)[:can_see_hidden_post]).to eq(true)
        expect(serialized_post_for_user(Fabricate(:user))[:can_see_hidden_post]).to eq(false)
      end
```